ksqldb-functional-tests/src/test/resources/query-validation-tests/row-id.json

```diff
@@ -0,0 +1,227 @@
+{
+  "comments": [
+    "Tests covering ROWID pseudocolumn"
+  ],
+  "tests": [
+    {
+      "name": "create ROWID sink user columns as STRINGs",
+      "properties": {
+        "ksql.rowid.enabled": true
+      },
+      "statements": [
+        "CREATE STREAM input (id INT KEY, val STRING, other_val STRING) WITH (kafka_topic='test_topic', value_format='JSON');",
+        "CREATE STREAM OUTPUT AS SELECT id, val AS ROWID FROM input;"
+      ],
+      "expectedException": {
+        "type": "io.confluent.ksql.util.KsqlStatementException",
+        "message": "`ROWID` is a reserved column name. You cannot use it as an alias for a column."
+      }
+    },
+    {
+      "name": "test ROWID sink user columns with pseudocolumn data types",
+      "properties": {
+        "ksql.rowid.enabled": true
+      },
+      "statements": [
+        "CREATE STREAM input (id INT KEY, val INTEGER, other_val BIGINT) WITH (kafka_topic='test_topic', value_format='JSON');",
+        "CREATE STREAM OUTPUT AS SELECT id, val AS ROWID FROM input;"
+      ],
+      "expectedException": {
+        "type": "io.confluent.ksql.util.KsqlStatementException",
+        "message": "`ROWID` is a reserved column name. You cannot use it as an alias for a column."
+      }
+    },
+    {
+      "name": "create ROWID source user columns as STRINGs",
+      "properties": {
+        "ksql.rowid.enabled": true
+      },
+      "statements": [
+        "CREATE STREAM input (id INT KEY, ROWID string) WITH (kafka_topic='test_topic', value_format='JSON');"
+      ],
+      "expectedException": {
+        "type": "io.confluent.ksql.util.KsqlStatementException",
+        "message": "'ROWID' is a reserved column name. You cannot use it as a name for a column."
+      }
+    },
+    {
+      "name": "test ROWID source user columns with pseudocolumn data types",
+      "properties": {
+        "ksql.rowid.enabled": true
+      },
+      "statements": [
+        "CREATE STREAM input (id INT KEY, ROWID INTEGER) WITH (kafka_topic='test_topic', value_format='JSON');"
+      ],
+      "expectedException": {
+        "type": "io.confluent.ksql.util.KsqlStatementException",
+        "message": "'ROWID' is a reserved column name. You cannot use it as a name for a column."
+      }
+    },
+    {
+      "name": "INSERT INTO ROWID should fail",
```

Comment:
Similar question regarding INSERT VALUES: do we want INSERT INTO ROWID to fail, or is it a valid way for users to interact with the record key?